### ISSUE TITLE
pytorch with ROCM on Windows

### ISSUE THREAD

**Issue by znsoftm** [pytorch/pytorch#106746]
### 🚀 The feature, motivation and pitch

pytorch with ROCM on Windows

Since the ROCM has supported Windows, when will the Pytorch be availabled on Windows?

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @vladimir-aubrecht @iremyux @Blackhex @cristianPanaite @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

**Comment by hongxiayang** [comment]
Windows Support for PyTorch is currently not available. This includes WSL2 based workflows.
AMD is continuing to invest in Windows support including the PyTorch ecosystem.
For details or future updates, please check: https://rocm.docs.amd.com/en/latest/rocm.html#rocm-on-windows

**Comment by shott92** [comment]
with ROCm 5.6 support on the nightly build, ROCm support from AMD on Windows 10 and 11 22h2, are there any technical blockers to PyTorch supporting ROCm on windows or is it just a development process?

**Comment by Yama-K** [comment]
> 

Don't think so, just matter of time for Pytorch to bake windows support with rocm.

**Comment by maxime-fleury** [comment]
They said they need MIOPEN to port pytorch

**Comment by NeedsMoar** [comment]
I just downloaded the latest version of AMD ProRender for Houdini and was surprised to find MIOpen.dll and RadeonML_MIOpen.dll sitting in the package's libs directory;  I'd expect they're not shipping an untested beta for a renderer for high end 3D software but there's no other indication of a Windows build anywhere else which is a shame.  I'm about ready to hate myself hard and go out and buy an A6000 or something (or just a 4090), especially since they're much more widely supported by and faster in renderers to begin with and I won't need to wait another 6 months to have generalized ML acceleration on the card AMD was selling with new ML acceleration instructions 10 months ago.  DirectML is practically unusable with the standard workflow all available stable diffusion software uses in python and conversion to jitting everything with native torch_directml or using the torch.onnx stuff is non-trivial and beyond my ability with unknown benefits.

Either AMD has two libraries called MIOpen or somebody has their wires crossed on getting a Windows build out to projects that need one to implement support, because who knows how long it'll be before a torch build with support is actually released after that happens.  I could always try generating an import lib for the DLL and doing it myself but I suspect nightmares lay down that route.  

First time I've been disappointed that I went with AMD, and every GPU I've had since 1999 was ATI then AMD.
